python/cudf/cudf/options.py

```diff
@@ -114,6 +115,40 @@ def set_option(name: str, val: Any):
     option.value = val
 
 
+class option_context(ContextDecorator):
+    """
+    Context manager to temporarily set options in the `with` statement context.
+
+    You need to invoke as ``option_context(pat, val, [(pat, val), ...])``.
+
+    Examples
+    --------
+    >>> from cudf import option_context
+    >>> with option_context('display.max_rows', 10, 'display.max_columns', 5):
+    ...     pass
+    """
+
+    def __init__(self, *args) -> None:
+        if len(args) % 2 != 0 or len(args) < 2:
```

Comment:
```suggestion
    def __init__(self, *args) -> None:
        if len(args) % 2 != 0:
```
Perhaps.

Rationale: this is a silly API because as a caller you have to special-case an empty options list if you want to use the context manager. That is, suppose you have some high-level code that accepts arbitrary options (including none), and you want them set when you are running some cudf/pandas code. You must do:

```python
options = ...
if len(options):
    manager = cudf.options_context(*options)
else:
   manager = contextmanager.nullcontext()
with manager:
    run_code
```

Rather than the much more natural.

```
with cudf.options_context(*options):
    ...
```

Reply:
For what it's worth -- I had some of the same suggestions/questions in mind as @wence- here (and in other comments) but I deferred to matching the `pandas` code. This behavior is identical to what pandas already has. I am fine with applying the suggestions, though.

Reply:
Yeah, this suggestion is forward-compatible with pandas usage, so I think it is fine to diverge.

Reply:
I changed the behavior of `with cudf.options_contex()` to reset all `cudf` options to default.

Reply:
Sorry for being unclear, I was posing that mostly as a hypothetical. I would recommend adhering to the same behavior as pandas. Just layer over what is already set, without clearing the user options that exist.

Reply:
No worries, reverted that change 👍